js/test/VCIClient.test.js

```diff
@@ -0,0 +1,93 @@
+import { DownloadFailedException } from "../src/exception/DownloadFailedException";
+import { NetworkRequestTimeoutException } from "../src/exception/NetworkRequestTimeoutException";
+import { requestCredential } from "../src/VCIClient";
+import nock from "nock";
+
+describe("VCI Client test", () => {
+  const mockCredentialResponse = {
+    format: "ldp_vc",
+    credential: {
+      issuanceDate: "2024-04-14T16:04:35.304Z",
+      credentialSubject: {
+        face: "data:image/jpeg;base64,/9j/goKCyuig",
+        dateOfBirth: "2000/01/01",
+        id: "did:jwk:eyJr80435=",
+        UIN: "9012378996",
+        email: "mockuser@gmail.com",
+      },
+      id: "https://domain.net/credentials/12345-87435",
+      proof: {
+        type: "RsaSignature2018",
+        created: "2024-04-14T16:04:35Z",
+        proofPurpose: "assertionMethod",
+        verificationMethod: "https://domain.net/.well-known/public-key.json",
+        jws: "eyJhbGciOiJSUzI11E_d-uifyj84U3XU8TA",
+      },
+      type: ["VerifiableCredential"],
+      "@context": [
+        "https://www.w3.org/2018/credentials/v1",
+        "https://domain.net/.well-known/context.json",
+        {
+          sec: "https://w3id.org/security#",
+        },
+      ],
+      issuer: "https://domain.net/.well-known/issuer.json",
+    },
+  };
+  const credentialEndpoint = "https://domain.net/credential";
+  const credentialAudience = "https://domain.net";
+  const downloadTimeoutInMillSeconds = 3000;
+  const credentialType = ["VerifiableCredential"];
+  const credentialFormat = "ldp_vc";
+  const proof = "eyJhbdFQwRnFWZjIcQku5nL1E_d-uifyj84U3XU8TA";
+  const accessToken = "eyJrahyw5Q_oeX7jXlHffwD2eBo3g";
+  const issuerMeta = {
+    credentialAudience,
+    credentialEndpoint,
+    downloadTimeoutInMillSeconds,
+    credentialType,
+    credentialFormat,
+  };
+
+  it("should make api call to credential endpoint with the right params in case of ldpVc", async () => {
+    nock("https://domain.net").post("/credential").reply(200, mockCredentialResponse);
```

Comment:
@adityankannan-tw Any specific reason on why we are using nock library for mocking purpose rather than jest which is also available?

Reply:
was facing issues with writing the timeout test scenario, so went with nock to maintain uniformity across tests.
Will see if i can figure out the timeout test scenario for some more time and update the code. Please share if you have any references for api timeout mocking also.

Reply:
Updated the tests to use jest.